File: src/in/co/trapps/jvsk/java/VariableExample.java
package in.co.trapps.jvsk.java;

/**
 * @author Akash Patra
 */
public class VariableExample {
    final int x = 1;
    int y = 2;

    /**
     * Read Only Variables (final) Initialization
     */
    public void initReadOnlyVariable() {
        final char a = 'A'; // Char type
        final int b = a; // Char to Int
        final String c = "Java"; // String

        System.out.println("Read Only Variable -> a: " + a + ", b: " + b + ", c: " + c);
    }

    /**
     * Mutable Variables Initialization
     */
    public void initMutableVariable() {
        char a = 'A'; // Char type
        int b = a; // Char to Int immediate assignment
        String c = "Java"; // String
        System.out.println("Mutable Variable -> a: " + a + ", b: " + b + ", c: " + c);

        a = 'B';
        b = a;
        c = "Java is a Champion";
        System.out.println("Mutable Variable (After Change) -> a: " + a + ", b: " + b + ", c: " + c);
    }

    /**
     * Access Instance Variable (val & var)
     */
    public void accessInstanceVariable() {
        System.out.println("Instance Variable -> x: " + x + ", y: " + y);
        y += 1; // Increment Mutable Variable
        System.out.println("Instance Variable (After Change) -> x: " + x + ", y: " + y);
    }
}

File: src/com/trapps/jvsk/kotlin/VariableExample.kt
package com.trapps.jvsk.kotlin

/**
 * @author Akash Patra
 */
class VariableExample {

    val x = 1
    var y = 2

    /**
     * Read Only Variables (val) Initialization
     */
    fun initReadOnlyVariable() {
        val a = 'A' // `Char` type is inferred
        val b: Int = a.toInt() // Char to Int immediate assignment
        val c: String = "Kotlin" // immediate assignment
        println("Read Only Variable -> a: $a, b: $b, c: $c")
    }

    /**
     * Mutable Variables (var) Initialization
     */
    fun initMutableVariable() {
        var a = 'A' // `Char` type is inferred
        var b: Int = a.toInt() // Char to Int immediate assignment
        var c = "Kotlin" // immediate assignment
        println("Mutable Variable -> a: $a, b: $b, c: $c")

        a = 'B'
        b = a.toInt()
        c = "Kotlin is evolving"
        println("Mutable Variable (After Change) -> a: $a, b: $b, c: $c")
    }

    /**
     * Access Instance Variable (val & var)
     */
    fun accessInstanceVariable() {
        println("Instance Variable -> x: $x, y: $y")
        y += 1 // Increment Mutable Variable
        println("Instance Variable (After Change )-> x: $x, y: $y")
    }
}